### PR TITLE
cmd/createtree: retry (with deadline) on Unavailable

### PR DESCRIPTION
**Pull request by daviddrysdale**
For #856; not a solution, but should make Travis less unreliable.

### CODE REVIEW

File: cmd/createtree/main.go [hunk 1]
```diff
@@ -72,15 +76,21 @@ func createTree(ctx context.Context) (*trillian.Tree, error) {
...
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("failed to dial %v: %v", *adminServerAddr, err)
 	}
 	defer conn.Close()
 
-	tree, err := trillian.NewTrillianAdminClient(conn).CreateTree(ctx, req)
-	if err != nil {
-		return nil, err
+	for {
```
Comment (codingllama) [inline cmd/createtree/main.go]: Personally I would push the retries to the bash scripts. Up to you.

File: cmd/createtree/main.go [hunk 2]
```diff
@@ -72,15 +76,21 @@ func createTree(ctx context.Context) (*trillian.Tree, error) {
...
+		tree, err := trillian.NewTrillianAdminClient(conn).CreateTree(ctx, req)
```
Comment (codingllama) [inline cmd/createtree/main.go]: nit: Create the client outside of the loop?

File: cmd/createtree/main.go (hunk 1 above)
Comment (daviddrysdale) [reply cmd/createtree/main.go]: Then there's no way to distinguish between error types -- `Unavailable` is the only one [documented as explicitly retriable](https://github.com/grpc/grpc-go/blob/master/codes/codes.go#L86).

File: cmd/createtree/main.go (hunk 2 above)
Comment (daviddrysdale) [reply cmd/createtree/main.go]: Done.

**Pull request merged**